dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -0,0 +1,75 @@
+class Api::V1::UserSchoolInfosController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_user_school_info
+
+  skip_before_action :verify_authenticity_token
+
+  # PATCH /api/v1/users_school_infos/<id>/update_last_confirmation_date
+  def update_last_confirmation_date
+    if @user_school_info.user.id == current_user.id
```

Comment:
isn't this covered by `authenticate_user`? If not, the preferred way to do this is with CanCan

Reply:
Updated code to use CanCan.